[DM-3216] Creating a Device Type With "ApplyConstraints" Can Now Be Used Without the "matchedNodeIds" List (#2629)
SoftwareAG/c8y-docs: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md

---
date: 

> title: Creating OPC UA device type without the matchedNodeIds list for apply constraints

Reviewer: ```suggestion
title: Creating OPC UA device type without the matchedNodeIds list for applyConstraints
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-3216
version: 10.20.97.0
---

> Previously, the device type creation via OPC UA management service API failed when `matchedNodeIds` list was not defined for `applyConstraints`. This issue 

Reviewer: ```suggestion
Previously, the device type creation via the OPC UA management service API failed when `matchedNodeIds` list was not defined for `applyConstraints`. This issue 
```

---

> date: '2024-11-13'

Reviewer: ```suggestion
date: 
```

> title: Creating a device type with auto-apply constraints now works without the matchedNodeIds List

Reviewer: ```suggestion
title: Creating a device type with auto-apply constraints now works without the matchedNodeIds list
```

> title: Creating a device type with auto-apply constraints now works without the matchedNodeIds list

Reviewer 2: ```suggestion
title: Creating OPC UA device type without the matchedNodeIds list for apply constraints
```

> title: Creating a Device Type With "ApplyConstraints" Can Now Be Used Without the "matchedNodeIds" List

Reviewer: ```suggestion
title: Creating a device type with auto-apply constraints now works without the matchedNodeIds List
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-Tf05_KQ-B
    label: OPC UA
build_artifact:
  - value: tc-MLn0oFRX-
    label: opcua
ticket: DM-3216
version: 10.20.97.0
---

> When creating a device type with "ApplyConstraints" and any properties, 

Reviewer: ```suggestion
Previously, when creating a device type with any auto-apply constraints properties set, 
```

> but without the "matchedNodeIds" list, the service would fail. This issue 

Reviewer: ```suggestion
but without the `matchedNodeIds` list, the service failed. This issue 
```

> has now been resolved, and "ApplyConstraints" can now be used without the 

Reviewer: ```suggestion
has now been resolved, and auto-apply constraints can now be used without the 
```

> "matchedNodeIds" list.

Reviewer: ```suggestion
`matchedNodeIds` list.
```

> Previously, when creating a device type with any auto-apply constraints properties set, 
> but without the `matchedNodeIds` list, the service failed. This issue 
> has now been resolved, and auto-apply constraints can now be used without the 
> `matchedNodeIds` list.

Reviewer 2: ```suggestion
Previously, the device type creation via OPC UA management service API failed when `matchedNodeIds` list was not defined for `applyConstraints`. This issue 
has now been resolved, and apply constraints can now be set without defining a 
`matchedNodeIds` list.
```
